Feature/rework aln variant metrics json

justinjj24 opened the pull request:
- Optional input aln/variant/aln+variant to workflow
   - aln (BAM/CRAM) and vcf inputs
   - aln input only
   - vcf input only
- Dropped Mutiqc module
- Container update v0.13.0
- NextflowVersion = >=23.10.1

mhebrard reviewed (changes requested):
1. To remove metrics.py.BAK
2. To remove metrics.py.UPDATE
3. Did you mention in a readme where the autosomes_non_gap_regions_bed come from / how is it generated ? 
4. To remove Dockerfile.BAK
5. To rename params.yaml.aln into params.aln.yaml
6. To rename params.yaml.vcf into params.vcf.yaml
7. about the output of the workflow. Currently it create 3 files (metrics.json, aln.metrics.json, variant.metrics.json)... A better solution would be to generate 1 file metrics.json witha nested structure. 
```
# if aln & vcf
"wgs_qc_metrics": {
        "aln_metrics": {
                "insert_size_std_deviation": "98.2",
                "mad_autosome_coverage": "0",
                "mean_autosome_coverage": "0.000644",
                "mean_insert_size": "447.3",
                "pct_autosomes_15x": "0.000016",
                "pct_reads_mapped": "99.63",
                "pct_reads_properly_paired": "97.9",
                "yield_bp_q30": "1996315"
        }
        "vcf_metrics": {
                "count_deletions": 848,
                "count_insertions": 850,
                "count_snvs": 8489,
                "ratio_heterozygous_homzygous_indel": 1.49,
                "ratio_heterozygous_homzygous_snv": 1.05,
                "ratio_insertion_deletion": 1.0,
                "ratio_transitions_transversions_snv": 2.13,
         }
}
```
If the aln is not provide, "aln_metrics" will be empty `{}`
If the vcf is not provide, "vcf_metrics" will be empty `{}`

justinjj24 commented:
> 1. To remove metrics.py.BAK
Removed.
> 2. To remove metrics.py.UPDATE
Removed.
> 3. Did you mention in a readme where the autosomes_non_gap_regions_bed come from / how is it generated ?
Yes, details provided in the README.md
> 4. To remove Dockerfile.BAK
Removed.
> 5. To rename params.yaml.aln into params.aln.yaml
Renamed.
> 6. To rename params.yaml.vcf into params.vcf.yaml
Renamed.
> 7. about the output of the workflow. Currently it create 3 files (metrics.json, aln.metrics.json, variant.metrics.json)... A better solution would be to generate 1 file metrics.json witha nested structure.
Right, in-progress!
> 
> ```
> # if aln & vcf
> "wgs_qc_metrics": {
>         "aln_metrics": {
>                 "insert_size_std_deviation": "98.2",
>                 "mad_autosome_coverage": "0",
>                 "mean_autosome_coverage": "0.000644",
>                 "mean_insert_size": "447.3",
>                 "pct_autosomes_15x": "0.000016",
>                 "pct_reads_mapped": "99.63",
>                 "pct_reads_properly_paired": "97.9",
>                 "yield_bp_q30": "1996315"
>         }
>         "vcf_metrics": {
>                 "count_deletions": 848,
>                 "count_insertions": 850,
>                 "count_snvs": 8489,
>                 "ratio_heterozygous_homzygous_indel": 1.49,
>                 "ratio_heterozygous_homzygous_snv": 1.05,
>                 "ratio_insertion_deletion": 1.0,
>                 "ratio_transitions_transversions_snv": 2.13,
>          }
> }
> ```
> 
> If the aln is not provide, "aln_metrics" will be empty `{}` If the vcf is not provide, "vcf_metrics" will be empty `{}`
This is less likely to implement in the current method.

mhebrard commented on tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/run.sh:
```diff
@@ -6,8 +6,9 @@
 NXF_ANSI_LOG=false
 
-nextflow run ../../main.nf \
+/home/ubuntu/bin/nf24.04.2/nextflow run ../../main.nf \
```
do not use the explicit path for your own nextflow installation. You should configure properly your environment to be able to run nextflow

```suggestion
nextflow run ../../main.nf \
```

mhebrard commented on tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/run.sh:
```diff
@@ -6,8 +6,9 @@
 	-config ../../nextflow.config -config ./NA12878-chr14-AKT1.config \
 	--inputs_list params.yaml \
 	-work-dir ./work \
+	-resume \
```
resume option should not be the default for the test
```suggestion
```

mhebrard commented on tests/NA12878_1000genomes-dragen-3.7.6/run.sh:
```diff
@@ -6,8 +6,9 @@
 NXF_ANSI_LOG=false
 
-nextflow run ../../main.nf \
+/home/ubuntu/bin/nf24.04.2/nextflow run ../../main.nf \
```
do not use the explicit path for your own nextflow installation. You should configure properly your environment to be able to run nextflow

```suggestion
nextflow run ../../main.nf \
```

mhebrard commented on tests/NA12878_1000genomes-dragen-3.7.6/run.sh:
```diff
@@ -6,8 +6,9 @@
 	-config ../../nextflow.config \
 	--inputs_list params.yaml \
 	-work-dir ./work \
+        -resume \
```
resume option should not be the default for the test
```suggestion
```

mhebrard reviewed (changes requested):
I would not include the *.aln.metrics.json & *.variant.metrics.json in the output_certified folder
